### DISCUSSION TITLE
Props drilling 문제가 생겼어요!

### DISCUSSION THREAD

**Discussion by yoonucho** [OttSideProject/OnDe-FE#8] [Ideas]
안녕하세요 😊

프로젝트에서 상태 관리는 기본적으로 **useState**만을 사용하려고 했었는데, 작업을 하다 보니 Props drilling 문제가 발생해서 이를 어떻게 해결할지 고민하던 중에 Zustand를 도입하는 게 좋을 것 같다는 생각이 들었습니다.

제가 컴포넌트를  쪼개다 보니 Props의  depth가  4단계까지 가는 상황이 발생하였습니다.  😹 

_Props drilling은 상태나 함수를 상위 컴포넌트에서 하위 컴포넌트로 전달할 때, 중간에 있는 컴포넌트들이 그 값을 직접 사용하지 않지만, 그저 전달만 하기 위해 props로 받아야 하는 상황을 말합니다.

이 방식은 컴포넌트 계층이 깊어질수록 복잡해지고, 유지보수가 어려워집니다. 상태나 함수가 필요한 컴포넌트에만 전달되지 않고 여러 단계의 컴포넌트에 불필요하게 전달되는 것이 문제입니다._

Props drilling에 대한 설명은 해당 포스팅을 첨부하겠습니다! https://velog.io/@rachel28/Prop-Drilling

[Zustand](https://zustand-demo.pmnd.rs/) 는 경량 상태 관리 라이브러리로, 전역 상태를 쉽게 관리할 수 있으면서도 리액트 컨텍스트나 다른 무거운 상태 관리 라이브러리보다 훨씬 간단하게 사용할 수 있습니다. 이를 통해 Props drilling 문제를 해결할 수 있고,코드의 가독성과 유지보수성을 높일 수 있을 것 같아요.

또한, Zustand는 최근 채용 공고에서도 자주 언급되는 라이브러리입니다. React와 함께 경량 상태 관리를 위해 요구되는 경우가 많아지고 있어, 저희가 이를 익히면 앞으로의 기술 스택에도 유리할 것 같습니다.

한번 검토해주시고, 의견 부탁드리겠습니다!

**Comment by yewon830** [comment]
zustand를 사용해봤는데 Redux보다 사용하기 간편하다고 느꼈습니다! 좋은 것 같습니다. 현업에서 어떻게 활용되는지는 잘 모르겠지만, 다양한 곳에서 쓰이는 유저 정보도 전역적으로 관리를 하면 편할 거 같아요!

**Comment by keemewoon** [comment]
저두 주스탠드로 관리하는게 좋을것같네요~^^

**Comment by yoonucho** [comment]
넵! 의견주셔서 감사합니당!
그러면 zustand를 설치해두고 환경설정을 해두겠습니당!

**Reply by yewon830** [reply]
수고하십니다 감사합니다!
